theurgy.lic

```diff
@@ -208,28 +223,54 @@ class TheurgyActions
     end
   end
 
-  def annotate_item_presence(data)
+  def last_favor_aspect
+    favor_re = /Your most recent favor was granted by (.*)\./
+    match = bput('favor', 'You are not currently favored', favor_re)
+    return unless favor_re.match(match)
+
+    last_favor_god = Regexp.last_match[1]
+    @immortal_to_aspect[last_favor_god]
+  end
+
+  def collect_item_info(data)
     data.each { |datum| datum[:items] ||= [] }
     items = data.map { |d| d[:items] }.flatten.uniq { |i| i[:name] }
-    item_presence = {}
-    items.each { |i| item_presence[i[:name]] = on_hand?(i) }
-
-    data.each do |datum|
-      datum[:items].each { |i| i[:on_hand] = item_presence[i[:name]] }
+    item_info = {}
+    items.each do |i|
+      item_count = count(i)
+      supply_levels = @settings.theurgy_supply_levels[i[:name]] || {}
+      item_info[i[:name]] = {
+        name: i[:name],
+        shop: i[:shop],
+        stackable: i[:stackable],
+        parts: (i[:shop] && i[:shop]['parts']) || i[:parts] || 1,
+        count: item_count,
+        on_hand: item_count > 0,
+        restock: item_count < (supply_levels['min'] || 1),
+        target: supply_levels['target'] || 1
+      }
     end
+
+    item_info
   end
 
   def rituals_remain?
     !@rituals.empty?
   end
 
-  def on_hand?(item)
-    if item[:name] == 'oil'
-      inside_container?('holy oil') || inside_container?('some oil')
-    elsif item[:any_container]
-      exists?(item[:name])
+  def count(item)
+    if item[:stackable]
+      if item[:any_container]
+        count_item_parts(item[:name])
+      else
+        count_item_parts("#{item[:name]} in my #{@theurgy_supply_container}")
+      end
     else
```

Comment:
this could be 
elseif item[:any_container]
  count_items(item[:name])
else
  count_items_in_container(item[:name], @theurgy_supply_container)
to shut up rubocop.

Reply:
Done.